objectstore/objectstore.go

```diff
@@ -0,0 +1,58 @@
+package objectstore
+
+import (
+	"io"
+
+	"github.com/fastly/compute-sdk-go/internal/abi/fastly"
+)
+
+type Entry struct {
+	io.Reader
+
+	validString bool
+	s           string
+}
+
+func (e *Entry) String() string {
+	if e.validString {
+		return e.s
+	}
+
+	// TODO(dgryski): replace with StringBuilder + io.Copy ?
+	b, err := io.ReadAll(e)
+	if err != nil {
+		return ""
+	}
+
+	e.validString = true
+
+	return string(b)
+
+}
+
+// Store represents a Fastly object store
+type Store struct {
+	objectstore *fastly.ObjectStore
+}
+
+// Open returns a handle to the named object store
+func Open(name string) (*Store, error) {
+	o, err := fastly.OpenObjectStore(name)
```

Comment:
Non-Blocking Comment:
We could provide more specific errors here if we wanted to:
- If the name is empty
- If the name is more than 255 characters
- If the name contains control characters
- If the requested store does not exist